remote-trait-object/src/forwarder.rs

```diff
@@ -0,0 +1,78 @@
+// Copyright 2020 Kodebox, Inc.
+// This file is part of CodeChain.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+use crate::port::Handler;
+use crate::service::ServiceHandler;
+use std::collections::HashMap;
+
+type ServiceId = String;
+
+pub struct ServiceForwarder {
+    service_handlers: HashMap<ServiceId, Box<dyn ServiceHandler + Send>>,
+}
+
+impl ServiceForwarder {
+    pub fn new() -> Self {
+        Self {
+            service_handlers: Default::default(),
+        }
+    }
+
+    pub fn add_service(&mut self, name: ServiceId, service: Box<dyn ServiceHandler + Send>) {
+        let insert_result = self.service_handlers.insert(name.clone(), service);
+        if insert_result.is_some() {
+            panic!("Duplicated service id {}", name);
+        }
+    }
+
+    pub fn forward(&self, input: String) -> String {
```

Comment:
It does more thing than just `forward` (the process of finding the target). In worst case someone might think that the returning `String` is some kind of key of target service object or path indicator, not the final result. How about `forward_and_call`?

Reply:
I think `forward` is giving enough information, but `forward_and_call` also looks good. I'll use it.